scripts/seed.py

```diff
@@ -0,0 +1,210 @@
+            challenge=challenge,
+            test_annotation=SimpleUploadedFile(fake.file_name(extension="txt"),
+                                               "1\n2\n3\n4\n5\n6\n7\n8\n9\n10", content_type="text/plain"),
+            codename="%s%d" % ("phase", i+1),
+        )
+        print "Challenge Phase created with \n name: %s \n challenge: %s" % (name, challenge.title)
+
+
+def create_leaderboard():
+    schema = {
+        'labels': ['score', ],
+        'default_order_by': 'score',
+    }
+    leaderboard = Leaderboard.objects.create(
+        schema=schema
+    )
+    print "Leaderboard created"
+    return leaderboard
+
+
+def create_dataset_splits(number_of_splits):
+    for i in range(number_of_splits):
+        global DATASET_SPLIT_ITERATOR
+        name = "Split %d" % (DATASET_SPLIT_ITERATOR + 1)
+        codename = "%s%d" % ('split', DATASET_SPLIT_ITERATOR + 1)
+        DatasetSplit.objects.create(
+            name=name,
+            codename=codename,
+        )
+        DATASET_SPLIT_ITERATOR += 1
+        print "Dataset Split created with \n name: %s \n codename: %s" % (name, codename)
+
+
+def create_challenge_phase_splits(challenge_phase, leaderboard, dataset_split):
+    ChallengePhaseSplit.objects.create(
+        challenge_phase=challenge_phase,
+        leaderboard=leaderboard,
+        dataset_split=dataset_split,
+        visibility=ChallengePhaseSplit.PUBLIC
+    )
+    print "Challenge Phase Split created with \n challenge_phase: %s \n dataset_split: %s" \
+          % (challenge_phase.name, dataset_split.name)
+
+
+def create_participant_team(user):
+    fake = Factory.create()
+    team_name = "%s Participant Team" % (fake.city())
+    team = ParticipantTeam.objects.create(
+        team_name=team_name,
+        created_by=user,
+    )
+    print "Participant Team created with \n team_name: %s \n created_by: %s" % (team_name, user.username)
+    Participant.objects.create(user=user, team=team, status="Self")
+    print "Participant created with \n user: %s \n team_name: %s" % (user.username, team_name)
+    return team
+
+
+print "Starting database seeder, Hang on :)"
+
+NUMBER_OF_CHALLENGES = 1
+NUMBER_OF_PHASES = 2
+NUMBER_OF_DATASET_SPLITS = 2
+DATASET_SPLIT_ITERATOR = 0
+
+create_admin_user()
+host_user = create_user(username="host")
+challenge_host_team = create_challenge_host_team(user=host_user)
+create_challenges(number_of_challenges=NUMBER_OF_CHALLENGES, host_team=challenge_host_team)
+
+challenges = Challenge.objects.all()
+for challenge in challenges:
+
+    # Create a leaderboard object for each challenge
+    leaderboard = create_leaderboard()
+
+    # Create Phases for a challenge
+    create_challenge_phases(number_of_phases=NUMBER_OF_PHASES)
+
+    # Create Dataset Split for each Challenge
+    create_dataset_splits(number_of_splits=NUMBER_OF_DATASET_SPLITS)
+    dataset_splits = DatasetSplit.objects.all().order_by('-created_at')[:NUMBER_OF_DATASET_SPLITS]
```

Comment:
Why are we selecting the datasets after they are just created. Can we simply just not return the object which are created in `create_dataset_splits` function. This actually can be discussed.

Same goes for ChallengePhase.